internal/controller/volumereplicationgroup_controller.go

```diff
@@ -1319,7 +1319,11 @@ func (v *VRGInstance) processAsPrimary() ctrl.Result {
 
 	v.reconcileAsPrimary()
 
-	if v.result.Requeue {
+	v.updateVRGDataReadyCondition()
```

Comment:
I am not sure this change is enough. The problem was that `reconcileAsPrimary` was returning `requeue` because of S3 upload issues, and that's fine if that's the only cause. But `reconcileAsPrimary` can also return requeue for other reasons. We shouldn't move on to restoring `KubeObjects` unless we're sure the only issue is with the S3 upload. Otherwise, we risk proceeding with restoring `KubeObjects` prematurely.

Reply:
@BenamarMk If the failure is something other than s3 upload, then datareadycondition will be false. We return on line 1327 if that is the case. This should be enough, right?

Reply:
Checked volume replication and the ConditionCompleted does mean that the image was promoted.
https://github.com/csi-addons/volume-replication-operator/blob/b77dac3d3523a0ae446bad6dc1c500647e46aed6/controllers/volumereplication_controller.go#L479